Added fruit shop implementation

DanaKhromenko opened the pull request.

RomanDubovskyi commented on src/main/java/dao/ShopOperationsDaoCsvImpl.java:
```diff
@@ -0,0 +1,66 @@
+import java.util.stream.Collectors;
+
+public class ShopOperationsDaoCsvImpl implements ShopOperationsDao {
+    private final String INPUT_FILE_NAME;
```
такий неймінг використовується тільки для констант (static final) змінні. 
```suggestion
    private final String inputFileName;
```
спробуй ще заранити локально mvn clean package, щоб перевірити код на чекстайл

DanaKhromenko replied:
О, блин, я просто mvn clean сделала...

RomanDubovskyi commented on src/main/java/validators/FruitDataValidatorCsvImpl.java:
```diff
@@ -0,0 +1,55 @@
+public class FruitDataValidatorCsvImpl implements FruitDataValidator {
+    private final String csvFilePath;
+
+    public FruitDataValidatorCsvImpl(String csvFilePath) {
```
Вхідні параметри краще передавати до методу одразу а не в конструктор - тоді не потрібно буде створювати новий об'єкт валідатора під файл

RomanDubovskyi commented on src/main/java/validators/FruitDataValidatorCsvImpl.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    @Override
+    public boolean validate() {
```
щоб зробити валідатор більш реюзабельним передавай в нього не файл а рядок або масив рядків. Валідатору має бути байдуже звідки прийшов контент (файл, ввід консолі чи http запит).

RomanDubovskyi commented on src/main/java/fruitOperationManagers/FruitOperationManagerCsvImpl.java:
```diff
@@ -0,0 +1,27 @@
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FruitOperationManagerCsvImpl implements FruitOperationManager {
```
це більш схоже на CsvFileService - так як ми тут просто читаємо файл, скіпаємо заголовок і віддаємо масив строк. можна переіменувати метод і закинути його в клас FileManagerCsvImpl

DanaKhromenko commented:
- fixed checkstyle;
- changed FruitDataValidatorCsvImpl-method signature and FruitDataValidator constructor to push fileName to method, not a constructor;
- changer validate-method's signature: now it takes List<String> but not filePath;
- merged FruitOperationManagerCsvImpl's method with FileManager.

RomanDubovskyi commented on src/main/java/dao/ShopOperationsDaoCsvImpl.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    @Override
+    public boolean validate() {
```
цей метод використовується в коді ?

RomanDubovskyi commented on src/main/java/validators/FruitDataValidatorCsvImpl.java:
```diff
@@ -0,0 +1,44 @@
+            return false;
+        }
+        // checking operation name
+        if (!data[0].equals("b")
```
це доречі буде незручно редачити якщо буде зростати кількість операцій, краще тоді зберегти в Set з їх назвою і перевіряти через метод contains
